osu.Game/Overlays/Music/Playlist.cs

```diff
@@ -46,9 +51,21 @@ public void Filter(FilterCriteria criteria)
 
             items.SearchTerm = criteria.SearchText;
             currentCriteria = criteria;
+
+            if (currentCriteria == criteria)
+                updateMusicControllerPlaylist();
+
+            items.FilterCompleted += () => Scheduler.AddOnce(updateMusicControllerPlaylist);
+
+            void updateMusicControllerPlaylist()
+            {
+                musicController.Playlist.Clear();
+                musicController.Playlist.AddRange(AllVisibleSets);
+            }
```

Comment:
So I was skeptical that this did anything, in particular that this will support reordering items correctly, and yet... it does? It does because every reorder of items in the list calls `OnItemsChanged()`, and every `OnItemsChanged()` for this container calls `Filter()`. And thus the list is cleared and re-populated anew on every item reorder.

Stress-testing it on a huge database (~230MB) it is pretty visible in there, although maybe not as much as I'd actually expect it to be after inferring the above:

![image](https://github.com/user-attachments/assets/38d7a4b9-f730-4e27-b638-1fc6abb6ff17)

@ppy/team-client am I the only one who has a problem with this? The code is otherwise (deceptively?) simple and it *appears* to work correctly...

Reply:
@bdach if you think it's okay then i'm fine with the simple approach until someone complains about performance.